src/ds/d500/d500-device.cpp

```diff
@@ -712,10 +712,17 @@ namespace librealsense
 
     bool d500_device::check_symmetrization_enabled() const
     {
-        command cmd{ ds::MRD, 0x80000004, 0x80000008 };
+        using namespace ds;
+        command cmd(GET_HKR_CONFIG_TABLE,
+            static_cast<int>(d500_calib_location::d500_calib_ram_memory),
+            static_cast<int>(d500_calibration_table_id::stream_pipe_config_id),
+            static_cast<int>(d500_calib_type::d500_calib_dynamic));
         auto res = _hw_monitor->send(cmd);
-        uint32_t val = *reinterpret_cast<uint32_t*>(res.data());
-        return val == 1;
+       
+        if (res.size() != sizeof(d500_stream_pipe_config_table))
+            throw std::runtime_error("Stream Config table has unexpected length");
```

Comment:
throw `invalid_value_exception`, instead - same as in `check_calib`

Reply:
Also, it's the stream_pipe_config table.
Not sure if this check is needed as `check_calib` is supposed to do everything?

Reply:
the check calib does not check the size of the payload - this if statement does it.

Reply:
Then I'd ask why
Check-calib checks the size, CRC and everything else matches.
If something's missing, we should ask if we need to add it there, and not add a check for size everywhere. I'm not sure it's important even in this instance....

Reply:
I don;t think this is relevant for this PR.
If you think we should improve the check_calib - maybe it would be better to open a ticket, and we will do it in another PR?